### PR TITLE
allow search over predicate sets from small to large

### PR DESCRIPTION
**Pull request by tomsilver**
and a few other minor changes, e.g., to printing

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -448,7 +448,8 @@ def _run_heuristic_search(
     check_goal: Callable[[_S], bool],
     get_successors: Callable[[_S], Iterator[Tuple[_A, _S, float]]],
     get_priority: Callable[[_HeuristicSearchNode[_S, _A]], Any],
-    max_expansions: int = 1000,
+    max_expansions: int = 10000000,
```
Comment (ronuchit) [inline src/utils.py:451]: Let's switch these to float("inf") as the default values?

File: src/utils.py [hunk 2]
```diff
@@ -531,14 +537,15 @@ def run_gbfs(
     check_goal: Callable[[_S], bool],
     get_successors: Callable[[_S], Iterator[Tuple[_A, _S, float]]],
     heuristic: Callable[[_S], float],
-    max_expansions: int = 1000,
+    max_expansions: int = 10000000,
```
Comment (ronuchit) [inline src/utils.py:540]: same change to float("inf")

File: src/approaches/grammar_search_invention_approach.py [hunk 3]
```diff
@@ -160,13 +160,32 @@ def _check_goal(s: FrozenSet[Predicate]) -> bool:
...
+            # Start the search with all of the candidates.
+            init = frozenset(candidates)
+        else:
+            assert CFG.grammar_search_direction == "smalltolarge"
+            # Successively consider larger predicate sets.
+            def _get_successors(s: FrozenSet[Predicate]
+                    ) -> Iterator[Tuple[None, FrozenSet[Predicate], float]]:
+                for predicate in sorted(set(candidates) - s):  # determinism
+                    # Actions not needed. Frozensets for hashing.
+                    yield (None, frozenset(s | {predicate}), 1.)
```
Comment (ronuchit) [inline src/approaches/grammar_search_invention_approach.py:180]: why are the costs always 1 instead of being the cost of removing/adding that predicate?

Comment (ronuchit) [reply src/approaches/grammar_search_invention_approach.py:180]: i mean: the cost of the predicate being added/removed, as produced by generate()

Comment (tomsilver) [reply src/approaches/grammar_search_invention_approach.py:180]: this is just like a "prior" on the predicate. those costs are factored in via true positives and false positives. is that what you mean?

**Pull request merged**